packages/components/src/components/data-table/_data-table-core.scss

```
@@ -103,7 +103,7 @@
   }
 
   .#{$prefix}--data-table th:last-of-type {
-    position: relative;
+    position: static; // Required to preserve border.
```

Super unimportant style-nit lol:

```suggestion
    // Required to preserve border
    position: static;
```